Myanmar: Separate names

ondenman opened the pull request:
There are multiple occurrences of two names being recorded in the name field. For example, in the data for the membership with the id of "59143f6e3cf2461789e4c78bc54afe74", the name of the person is: "U Phye Ral (or) U Myint Than Htoo”. ( http://api.openhluttaw.org/en/memberships?page=2 )

This commit separates the two parts of the string at 'or'. The first part is treated as the 'official' name. The second part is stored in `:other_name`.

ondenman committed:
Separate multiple names

The source does not store aliases in an 'other_names' field. Instead,
it stores aliases all in one string, separated by (or).

We take the assumption that the first name is the 'official' name.
(Open Huttlaw has been asked whether the order of the names has
any significance. Awaiting reply.) The second name is stored as
an 'other_name'.

ondenman committed:
Tidy name strings

Since we're splitting these names from an existing string, we
have some whitespace that needs to be dealt with. String#tidy
does this.

tmtmtmtm commented on lib/member_formatter.rb:
```diff
@@ -1,5 +1,11 @@
+  end
+end
+
 class MemberFormatter
```
I don't really understand what this class is. In what way is it a 'Formatter'?

tmtmtmtm commented on lib/member_formatter.rb:
```diff
@@ -12,7 +18,11 @@ def initialize(member)
+  end
+
+  field :other_name do
+    names[1].tidy unless names[1].nil?
```
What if `names` has more than two elements?

tmtmtmtm commented on lib/member_formatter.rb:
```diff
@@ -12,7 +18,11 @@ def initialize(member)
+    names[0].tidy
+  end
+
+  field :other_name do
```
This should probably be `:alternate_names`, so that CSV-to-Popolo will add a `note`, rather than leaving it bare.

tmtmtmtm commented on lib/member_formatter.rb:
```diff
@@ -12,7 +18,11 @@ def initialize(member)
 
   field :name do
-    person[:name]
+    names[0].tidy
```
Would be a little cleaner to use `.first` here rather than `[0]`

ondenman committed:
Use #first instead of [0]

Looks tidier

ondenman committed:
Rename :other_name field to :alternate_names

CSV-to-Popolo records alternate names with a note: 'alternate name'
which would make it clear that the names we are storing here
are aliases.

ondenman committed:
Handle multiple alternate names

Can now record a single alternate name or many.

Records multiple alternate names separated by a ';'.

ondenman committed:
Rename MemberFormatter to Member

Member is a better representation of what this class represents:
a member of the legislature.

ondenman committed:
Remove Members#format

Moves the logic -- creating an array of Member objects --
into #to_a.

The naming of #format was confusing and since the purpose
of Members is to return a list of Member objects, the code
is better off living in #to_a.

tmtmtmtm commented on lib/member.rb:
```diff
@@ -12,7 +18,11 @@ def initialize(member)
+  end
+
+  field :alternate_names do
+    names.drop(1).map(&:tidy).join(';') unless names[1].nil?
```
would probably be simpler and clearer to `to_a` these first rather than testing `names[1]`

tmtmtmtm commented on lib/members.rb:
```diff
@@ -11,7 +11,9 @@ def initialize(url)
   end
 
   def to_a
```
I think you should fix up some of the issues with the existing scraper first — e.g. replacing this `to_a` with a `members` field on a serialized MembersPage class — then come back to making this PR.

tmtmtmtm reviewed (changes requested):
NB: You're also still 'require'-ing the Member class under its old name.

ondenman commented:
Closing in favour of: https://github.com/everypolitician-scrapers/myanmar-lower-house-2016/pull/4
